lib/rails_config_validator/validator.rb

```diff
@@ -0,0 +1,83 @@
+require 'kwalify'
+require 'yaml'
+require 'rails_config_validator/errors'
+
+module RailsConfigValidator
+  class Validator
+    def initialize(config_path, env, schema_path: nil)
+      @config_path = config_path
+      @schema_path = schema_path || config_to_schema_path(config_path)
+      @env = env
+    end
+
+    def valid?
+      meta_validate.empty? && validate.empty?
+    rescue
+      # ignored
+      false
+    end
+
+    def valid!
+      mv = meta_validate
+      fail InvalidSchemaError,
+           "Incorrect schema #{schema_path}; errors: #{mv}" unless mv.empty?
+      v = validate
+      fail InvalidConfigurationError,
+           "Incorrect config #{config_path}; errors: #{v}" unless v.empty?
+    end
+
+    def meta_validate
+      meta_validator = Kwalify::MetaValidator.instance
+      errors = meta_validator.validate(schema)
+      errors.map do |error|
+        "[#{error.path}](#{error.error_symbol}) #{error.message}"
+      end
+    rescue Errno::ENOENT => e
+      raise InvalidSchemaError, e
```

Comment:
Isn't the exception name misleading then? I see there's a MissingConfigurationError class, maybe it would make sense to use MissingSchemaError here.